apps/native/screens/OwnerMapScreen.tsx

```diff
@@ -0,0 +1,126 @@
+import React, {useEffect, useState} from 'react';
+import {MapComponent} from "@ui/MapComponent";
+import {db} from '../../web/firebase.js';
+import {View, StyleSheet, ActivityIndicator, TouchableOpacity, Text} from 'react-native';
+import {collection, getDocs, query} from "firebase/firestore";
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import FontAwesome5 from "react-native-vector-icons/FontAwesome5";
+
+export function OwnerMapScreen({ navigation }: any) {
+
+    const [uID, setUserId] = useState("")
+    const [ownedUnits, setOwnedUnits] = useState([]);
+    const [isLoading, setIsLoading] = useState(true);
+
+    useEffect(() => {
+        const fetchUserId = async () => {
+            // You'll see that userId is logged in console (just not in time for fetchData function)
+            await AsyncStorage.getItem('userUID')
+                .then((value) => {
+                    const data = JSON.parse(value);
+                    console.log('JSON data: ' + data)
+                    setUserId(data);
+                });
+        }
```

Comment:
Nice job here with the asyncstorage, I think this is how everyone should be checking this the same way.